src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final String EMAIL = "bob@gmail.com";
+    private static final String PASSWORD = "1234";
+    private static final String ENCODED_PASSWORD = "111111111";
+    private static final Set<Role> ROLES = Set.of(new Role(Role.RoleName.USER));
+    private static final long ID = 1L;
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private AuthenticationService auth;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        auth = new AuthenticationServiceImpl(userService, roleService, passwordEncoder);
+        user = new User(ID, EMAIL, PASSWORD, ROLES);
+    }
+
+    @Test
+    void register_Ok() {
+        Mockito.when(roleService.getRoleByName(Role.RoleName.USER.name()))
+                .thenReturn(new Role(Role.RoleName.USER));
+        Mockito.when(userService.save(Mockito.any())).thenReturn(user);
+
+        User actual = auth.register(EMAIL, PASSWORD);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(ID, actual.getId());
+        Assertions.assertEquals(EMAIL, actual.getEmail());
+        Assertions.assertEquals(PASSWORD, actual.getPassword());
+        Assertions.assertEquals(ROLES, actual.getRoles());
+    }
+
+    @Test
+    void register_RoleNotFound() {
+        Mockito.when(roleService.getRoleByName(Role.RoleName.USER.name())).thenReturn(null);
+
+        Assertions.assertThrows(NullPointerException.class,
+                () -> auth.register(EMAIL, PASSWORD), "Role not found");
+    }
+
+    @Test
+    void login_Ok() {
+        user.setPassword(ENCODED_PASSWORD);
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(user));
+        Mockito.when(passwordEncoder.matches(PASSWORD, ENCODED_PASSWORD)).thenReturn(true);
+        User actual = null;
+        try {
+            actual = auth.login(EMAIL, PASSWORD);
+        } catch (AuthenticationException e) {
+            Assertions.fail();
+        }
```

Comment:
it's ok to make a test method throw an exception. No need to catch it here as you mock the responses you should write the test in such a way that exceptions can't occur